cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/WindowAggregationDefinition.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Specifies how a window aggregation operation should be executed.
+ */
+public class WindowAggregationDefinition extends AggregationDefinition {
+  List<Expression> partitionExpressions;
+  private WindowAggregationDefinition(List<Expression> groupByExpressions,
+                                      Map<String, Expression> selectExpressions,
+                                      List<Expression> partitionExpressions) {
+    super(groupByExpressions, selectExpressions);
+    this.partitionExpressions = partitionExpressions;
+  }
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * @return A builder to create a WindowAggregationDefinition.
+   */
+  public List<Expression> getPartitionExpressions() {
+    return partitionExpressions;
+  }
+
+  /**
+   * Builds a WindowAggregationDefinition using fields to partition  and fields to select.
+   * The fields to select must be specified.
+   */
+  public static class Builder {
+    private List<Expression> groupByExpressions;
```

Comment:
If `groupByExpressions` is always going to be an empty list, it doesn't need to be a field inside the builder. You can directly invoke the constructor with an empty list instead.

Reply:
Removed group by expressions and extending from Aggregation Definition.Code refactoring  can be done after all the pushdowns are completed